lib/tether/ops_linux.go

```diff
@@ -490,6 +492,24 @@ func updateDefaultRoute(t Netlink, link netlink.Link, endpoint *NetworkEndpoint)
 	}
 
 	if bTablePresent {
+		// Gateway IP may not container network mask, so it is taken from the assigned interface configuration
+		// where network mask has to be defined.
+		gwNet := &net.IPNet{
+			IP:   gw.IP.Mask(newIP.Mask),
+			Mask: newIP.Mask,
+		}
+
+		log.Debugf("Adding route to default gateway network: %s/%s", gwNet.IP, gwNet.Mask)
+
+		route = &netlink.Route{LinkIndex: link.Attrs().Index, Dst: gwNet, Table: bridgeTableNumber}
+		if err := t.RouteAdd(route); err != nil {
+			if errno, ok := err.(syscall.Errno); !ok || errno != syscall.EEXIST {
+				return fmt.Errorf(
```

Comment:
let's not wrap this - it makes the logic flow less obvious and doesn't gain anything in modern environments.

Reply:
It actually does in case of DHCP, if IP address changes within the same subnet it will case and error.